lib/sql-string.js

```diff
@@ -35,6 +35,11 @@ function escape(val, timeZone, dialect, format) {
     break;
   }
 
+  // moment
+  if (val.toDate) {
```

Comment:
Yes, we can't (if its not moment). Its all up to implementor what `toDate` might return.